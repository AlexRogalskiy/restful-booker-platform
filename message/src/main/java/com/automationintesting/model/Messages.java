package com.automationintesting.model;

import com.fasterxml.jackson.annotation.JsonProperty;

import java.util.ArrayList;
import java.util.List;

public class Messages {

    @JsonProperty
    private List<MessageSummary> messages;

    public Messages(List<MessageSummary> messages) {
        this.messages = messages;
    }

    public List<MessageSummary> getMessages() {
        return messages;
    }

}
