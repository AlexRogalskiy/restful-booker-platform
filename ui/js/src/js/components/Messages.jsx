import React from 'react';
import { API } from '../libs/Api.js';

export default class Messages extends React.Component {

    constructor() {
        super();

        this.state = {
            messages : []
        }

        this.deleteMessage = this.deleteMessage.bind(this);
        this.refreshMessageList = this.refreshMessageList.bind(this);
    }

    componentDidMount() {
        this.refreshMessageList()
    }

    deleteMessage(id) {
        API.deleteMessage(id, this);
    }

    refreshMessageList(){
        API.getMessages(this);
    }

    render() {
        return <div class="messages">
            <div className="row">
                <div className="col-sm-2 rowHeader"><p>Name</p></div>
                <div className="col-sm-9 rowHeader"><p>Subject</p></div>
                <div className="col-sm-1"></div>
            </div>
            {this.state.messages.map((value, index) => {
                return  <div className="row detail" key={index}>
                            <div className="col-sm-2"><p>{value.name}</p></div>
                            <div className="col-sm-9"><p>{value.subject}</p></div>
                            <div className="col-sm-1">
                            <span className="fa fa-remove roomDelete"></span>
                            </div>
                        </div>
            })}
        </div>
    }

}