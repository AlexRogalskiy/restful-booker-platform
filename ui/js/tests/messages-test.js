import React from 'react';
import Messages from '../src/js/components/Messages.jsx';
import nock from 'nock';
import { waitForState } from 'enzyme-async-helpers';

beforeAll(() => {
    nock('http://localhost')
        .persist()
        .get('/message')
        .reply(200, {
                messages : [
                    {
                        "name" : "Mark Winteringham",
                        "subject" : "Subject description here"
                    }, {
                        "name" : "James Dean",
                        "subject" : "Another description here"
                    }, {
                        "name" : "Janet Samson",
                        "subject" : "Lorem ipsum dolores est"
                    }
                ]
            }
        );
});

test('Renders the list of messages correctly', async () => {
    const messageComponent = shallow(<Messages />);

    await waitForState(messageComponent, state => state.messages.length === 3);

    expect(messageComponent).toMatchSnapshot();
});

test('Deletes message when selected to delete', (done) => {
    let messageDeleteMock = nock('http://localhost')
                                .delete('/message/1')
                                .reply(201, () => {
                                    done();
                                });

    const messageComponent = shallow(<Messages />);
    messageComponent.instance().deleteMessage(1);

    let didNockAcceptRequest = messageDeleteMock.isDone();
    expect(didNockAcceptRequest).toBe(true);
});