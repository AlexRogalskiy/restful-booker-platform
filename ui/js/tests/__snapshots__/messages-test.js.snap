// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders the list of messages correctly 1`] = `
<div
  class="messages"
>
  <div
    className="row"
  >
    <div
      className="col-sm-2 rowHeader"
    >
      <p>
        Name
      </p>
    </div>
    <div
      className="col-sm-9 rowHeader"
    >
      <p>
        Subject
      </p>
    </div>
    <div
      className="col-sm-1"
    />
  </div>
  <div
    className="row detail"
    key="0"
  >
    <div
      className="col-sm-2"
    >
      <p>
        Mark Winteringham
      </p>
    </div>
    <div
      className="col-sm-9"
    >
      <p>
        Subject description here
      </p>
    </div>
    <div
      className="col-sm-1"
    >
      <span
        className="fa fa-remove roomDelete"
      />
    </div>
  </div>
  <div
    className="row detail"
    key="1"
  >
    <div
      className="col-sm-2"
    >
      <p>
        James Dean
      </p>
    </div>
    <div
      className="col-sm-9"
    >
      <p>
        Another description here
      </p>
    </div>
    <div
      className="col-sm-1"
    >
      <span
        className="fa fa-remove roomDelete"
      />
    </div>
  </div>
  <div
    className="row detail"
    key="2"
  >
    <div
      className="col-sm-2"
    >
      <p>
        Janet Samson
      </p>
    </div>
    <div
      className="col-sm-9"
    >
      <p>
        Lorem ipsum dolores est
      </p>
    </div>
    <div
      className="col-sm-1"
    >
      <span
        className="fa fa-remove roomDelete"
      />
    </div>
  </div>
</div>
`;
